test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp

```diff
@@ -93,30 +105,62 @@ class ImageInventoryTest : public CppUnit::TestFixture
 		StandardTestEnumerateInstances<mi::Container_ImageInventory_Class_Provider>(m_keyNames, context, CALL_LOCATION(errMsg));
 
 		// Get images using command line
-		CPPUNIT_ASSERT(!system("docker images -q --no-trunc > /tmp/docker_image_ids.txt"));
+		CPPUNIT_ASSERT(!system("docker images --no-trunc > /tmp/docker_image_ids.txt"));
 
 		FILE* idFile = fopen("/tmp/docker_image_ids.txt", "r");
 		CPPUNIT_ASSERT(idFile);
 
-		wchar_t id[65];
+		wchar_t line[200];
 		set<wstring> allIds;
 
-		// Full image IDs (one per line)
-		while (fwscanf(idFile, L"%ls", id) != EOF)
+		int matched = 0;
+
+		// Full image info (header plus one per line)
+		if (fgetws(line, 200, idFile) != NULL)
 		{
-			allIds.insert(wstring(id));
+			// parse header to get column offsets
+			wstring hdrstr(line);
+			string::size_type repoix = hdrstr.find(L"REPOSITORY", 0);
+			CPPUNIT_ASSERT(repoix != wstring::npos);
+			string::size_type tagix = hdrstr.find(L"TAG", 0);
+			CPPUNIT_ASSERT(tagix != wstring::npos);
+			string::size_type idix = hdrstr.find(L"IMAGE ID", 0);
+			CPPUNIT_ASSERT(idix != wstring::npos);
+			string::size_type createdix = hdrstr.find(L"CREATED", 0);
+			CPPUNIT_ASSERT(createdix != wstring::npos);
+
+			while (fgetws(line, 200, idFile) != NULL)
+			{
+				wstring linestr(line);
+				wstring id = substrTrim(linestr, idix, createdix);
+				bool match = false;
+				for (unsigned i = 0; i < context.Size(); ++i)
+				{
+					// Verify the InstanceID
+					if (id.compare(context[i].GetKey(L"InstanceID", CALL_LOCATION(errMsg))) == 0)
+					{
+						match = true;
+						matched++;
+						break;
+					}
+				}
+
+				if (!match)
+				{
+					// not found. If the repo & tag are both "<none>", then this is ok since enumerate may filter these
+					wstring none(L"<none>");
+					wstring repo = substrTrim(linestr, repoix, tagix);
+					CPPUNIT_ASSERT_EQUAL(none, repo);
```

Comment:
But this also makes the CPPUNIT_ASSERT_EQUAL less meaningful. I'd rather have one duplicated string (which, by the way, would be optimized away) and make the CPPUNIT_ASSERT_EQUAL test output more meaningful.